### PR TITLE
20.3 fb publication data 7128

### PR DESCRIPTION
**Pull request by labkey-klum**
#### Rationale
Add support for non-integrated publication data to be linked and displayed in the publication details view. The component will be similar to what is rendered for study non-integrated data and will allow users to download the related data.

#### Changes
* add a new junction table to track publications to documents
* update the import ETL to populate the junction table and queries
* add the new component to render the available documents

### CODE REVIEW

File: resources/schemas/dbscripts/postgresql/cds-20.002-20.003.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+CREATE TABLE cds.PublicationDocument
+(
+    Publication_Id VARCHAR(250) NOT NULL,
+    Document_Id VARCHAR(250) NOT NULL,
+    Container ENTITYID NOT NULL,
+
+    CONSTRAINT PK_PublicationDocument PRIMARY KEY (Publication_Id, Document_Id, Container),
+    CONSTRAINT FK_PublicationId_PublicationId FOREIGN KEY (Container, Publication_Id) REFERENCES cds.Publication (Container, Id),
+    CONSTRAINT FK_DocumentId_DocumentId FOREIGN KEY (Container, Document_Id) REFERENCES cds.Document (Container, Document_Id)
+);
```
Comment (labkey-bpatel) [inline resources/schemas/dbscripts/postgresql/cds-20.002-20.003.sql:23]: Maybe beneficial to add an Index to the FKs defined (though I don't see Indexes defined for other FKs in the older upgrade scripts, so maybe there's a reason behind that?).

File: webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js [hunk 2]
```diff
@@ -117,36 +112,27 @@ Ext.define('Connector.view.module.NonIntegratedDataAvailability', {
...
+        console.warn('need to override getData')
+    },
+
     hasContent : function() {
-        var reports = this.initialConfig.data.model.data.non_integrated_assay_data;
+        console.log('has content');
+
+        // todo delete this unused code
+        var reports = this.getData();
         if (reports) {
```
Comment (labkey-bpatel) [inline webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js]: Looks like a "todo" still needs to be handled or the maybe just the comment needs to be removed.

File: webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js [hunk 3]
```diff
@@ -117,36 +112,27 @@ Ext.define('Connector.view.module.NonIntegratedDataAvailability', {
         this.callParent();
     },
 
+    getData : function() {
+        console.warn('need to override getData')
+    },
+
     hasContent : function() {
-        var reports = this.initialConfig.data.model.data.non_integrated_assay_data;
+        console.log('has content');
```
Comment (labkey-bpatel) [inline webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js]: Is console.log() needed here?

File: webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js [hunk 4]
```diff
@@ -174,53 +160,152 @@ Ext.define('Connector.view.module.NonIntegratedDataAvailability', {
...
-                    '</li>',
+                '<li class="non-integrated-data-li">',
+                '<a href="#learn/learn/Assay',
+                '/{assayIdentifier}">',
+                '{label:htmlEncode}',
+                '</a>',
+                '&nbsp;{suffix}&nbsp;',
+                '<a href="{filePath}" target="_blank"><img alt="{assayIdentifier}" src="' + LABKEY.contextPath + '/Connector/images/download-icon.svg' + '" height="13" width="13" align="left"/></a>',
+                '</li>',
                 '</ul>',
```
Comment (labkey-bpatel) [inline webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js]: Looks like indentation here (lines 203-210) and in the other places in this function is lost.

**Review by labkey-bpatel — APPROVED** [review]
Looks great overall, please review my comments for minor changes.

File: webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js (hunk 2 above)
Comment (labkey-klum) [reply webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js]: yes, done

File: webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js (hunk 4 above)
Comment (labkey-klum) [reply webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js]: ooh weird, IntelliJ must have done that on copy/paste my bad!

**Pull request merged**